app/weeklyPicks/page.tsx

```diff
@@ -79,31 +79,33 @@ export default function WeeklyPickForm() {
     resolver: zodResolver(FormSchema),
   });
 
-  const onSubmit = useCallback(
-    async (data: z.infer<typeof FormSchema>) => {
-      try {
-        const teamSelect = data.type.toLowerCase();
-        localStorage.setItem('team', data.type);
-
-        const teamID = NFLTeams.find(
-          (team) => team.teamName.toLowerCase() === teamSelect,
-        )?.$id;
-        const resultJSON = `${user.id}:{"team":"${teamID}","correct":true}`;
-        const appendNewResult = allPicks
-          ? `{${allPicks},${resultJSON}}`
-          : `{${resultJSON}}`;
-
-        await createWeeklyPicks({
-          gameId: '66311a210039f0532044',
-          gameWeekId: '6622c7596558b090872b',
-          userResults: appendNewResult,
-        });
-      } catch (error) {
-        console.error('Submission error:', error);
-      }
-    },
-    [NFLTeams, user, allPicks],
-  );
+  const onSubmit = async (data: z.infer<typeof FormSchema>) => {
+    try {
+      const teamSelect = data.type.toLowerCase();
+      localStorage.setItem('team', data.type);
+
+      const teamID = NFLTeams.find(
+        (team) => team.teamName.toLowerCase() === teamSelect,
+      )?.$id;
+
+      // parse the user picked result
+      const userPick = JSON.parse(
+        `{"${user.id}":{"team":"${teamID}","correct":true}}`,
```

Comment:
m: Make a new function that takes in 2 props. then the return is this new object. This is very prone to error unless we are able to isolate a test for it.

Reply:
✅ Done!